spec/support/git_spec_helpers.rb

```diff
@@ -23,6 +23,41 @@ def repo(options = {})
     end
   end
 
+  # Creates a local repository in 'local-repo' folder with a remote repository
+  # which points to 'some-repo' folder. The current working directory is set to
+  # the local repository's root directory.
+  def with_remote_repo
+    repo do
+      touch 'update-me'
+      `echo update >> update-me`
+      touch 'delete-me'
+      `echo delete >> delete-me`
+      `git add . 2>&1 > #{File::NULL}`
+      `git commit -m "Initial commit" 2>&1 > #{File::NULL}`
+
+      Dir.chdir '..' do
+        remote_url = "file://#{File.expand_path('some-repo')}"
+        `git clone "#{remote_url}" local-repo 2>&1 > #{File::NULL}`
+      end
+
+      Dir.chdir '../local-repo' do
+        yield if block_given?
+      end
+    end
+  end
+
+  # Retrieve sha1 based on git ref
+  #
+  # @param ref [String] git ref
+  # @return [String] ref's sha1
+  def get_sha1(ref)
+    `git show-ref --head`.
+      split("\n").
+      find { |x| x =~ /#{Regexp.quote(ref)}/i }.
```

Comment:
Another danger here is you aren't anchoring your regex with `^` / `$`, so a ref of `foo` will match `foo`, `foobar`, `somefoobar`, etc. Fortunately we should be able to avoid entirely with `git rev-parse ...`.

Reply:
👍🏻